src/fees/saga.ts

```diff
@@ -150,6 +155,26 @@ export function* estimateSendFee(tokenAddress: string) {
   return feeInfo
 }
 
+export function* estimateSwapFee(tokenAddress: string) {
+  const tx: CeloTransactionObject<any> = yield call(
+    buildSendTx,
+    tokenAddress,
+    PLACEHOLDER_AMOUNT,
+    PLACEHOLDER_ADDRESS,
+    PLACEHOLDER_COMMENT
+  )
+
+  // TODO: calculate the fee accurately.
```

Comment:
how would that work? i thought you need a buy or sell amount to query the `approveSwap` endpoint, and we can't get the sell amount without calculating the fee first 🙈

Reply:
I was thinking in something like, calling the approveSwap endpoint with the max amount as sell amount, but just to have an estimation on the gas. Then we have to query `approveSwap` again with the max amount - fees.
I don't like this option though, since it takes a relevant amount of time to query 0x.